deploy: support rook deployment for v1.8+

Madhu-1 opened the pull request:
In Rook v1.8+ the path for the deployment articafts  are changed from `"cluster/examples/kubernetes/ceph`  to `deploy/examples`.
    
Fixes: #2787

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: re-alignment of code in rook.sh

currently, rook.sh code is not aligned properly
unside the functions, this commit does
the code alignment.

PS: this is done by vscode for me :)

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: support rook deployment for v1.8+

In Rook v1.8+ the path for the deployment articafts
are changed from `"cluster/examples/kubernetes/ceph`
to `deploy/examples`.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: update rook version to 1.8.2

update rook version to 1.8.2 in build.env
to pick up the latest rook release.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

yati1998 reviewed (commented):
A small typo in commit message :  `deploy/exaples` -> `deploy/examples`

yati1998 reviewed (dismissed):
LGTM

Madhu-1 commented:
/retest  all

Madhu-1 commented:
@mergifyio rebase

Madhu-1 commented:
/retest ci/centos/k8s-e2e-external-storage/1.21

Madhu-1 commented:
/test ci/centos/upgrade-tests-rbd

Madhu-1 commented:
>ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@n29.dusty.ci.centos.org 'podman pull --authfile=~/.podman-auth.json registry-ceph-csi.apps.ocp.ci.centos.org/rook/ceph:v1.8.2 && podman tag registry-ceph-csi.apps.ocp.ci.centos.org/rook/ceph:v1.8.2 rook/ceph:v1.8.2 docker.io/rook/ceph:v1.8.2'
Warning: Permanently added 'n29.dusty.ci.centos.org,172.19.2.93' (ECDSA) to the list of known hosts.
Trying to pull registry-ceph-csi.apps.ocp.ci.centos.org/rook/ceph:v1.8.2...
Error: initializing source docker://registry-ceph-csi.apps.ocp.ci.centos.org/rook/ceph:v1.8.2: reading manifest v1.8.2 in registry-ceph-csi.apps.ocp.ci.centos.org/rook/ceph: manifest unknown: manifest unknown

script returned exit code 125

@nixpanic looks like the image is still not available in the registry. do i need to rebuild anything manually in CI?

nixpanic commented:
Mirroring images is done daily, see https://github.com/ceph/ceph-csi/blob/ci/centos/mirror/mirror-cronjob.yaml

If required, we can manually run such a job. Easier to wait a day :wink:

Madhu-1 commented:
@mergifyio rebase

nixpanic commented:
@Madhu-1 looks like some other change is needed?
```
MountVolume.SetUp failed for volume "mon-endpoint-volume" : configmap "rook-ceph-mon-endpoints" not found
```
https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/k8s-e2e-external-storage-1.21/detail/k8s-e2e-external-storage-1.21/2455/pipeline

Madhu-1 commented:
> @Madhu-1 looks like some other change is needed?
> 
> ```
> MountVolume.SetUp failed for volume "mon-endpoint-volume" : configmap "rook-ceph-mon-endpoints" not found
> ```
> 
> https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/k8s-e2e-external-storage-1.21/detail/k8s-e2e-external-storage-1.21/2455/pipeline

Yes @nixpanic testing it locally. will update once i found what is changed

Madhu-1 committed:
deploy: log complete ouput of rook operator

kubectl log with labels will log only
last 10 lines by default adding tail=-1
to log complete output.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: add check for device-health-metrics pool

Rook creates a detault pool with name
device_health_metrics for
device-health-metrics CephBlockPool CR.

device-health-metrics is added to cluster-test.yaml
in Rook 1.8

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
local testing passed for me

```
Wait for rook deploy... 225s
Checking CEPH cluster state: [Created]
Creating CEPH cluster is done. [HEALTH_OK]

Checking MDS (myfs) stats... 0s
MDS (myfs) active_count: [1]
Filesystem (myfs) is successfully created...

Checking RBD (device_health_metrics) stats... 0s
RBD (device_health_metrics) is successfully created...
```

Madhu-1 commented:
Looks like something wrong is going on in CI logs are available at https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/upgrade-tests-cephfs/detail/upgrade-tests-cephfs/4449/pipeline/99 and https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/upgrade-tests-cephfs/detail/upgrade-tests-cephfs/4449/pipeline/86

@travisn @leseb currently we using deploying Rook 1.6.2, everything is working file, In this PR am moving to latest Rook version  1.8.2 but looks like things are not stable, any idea what could be wrong here?

Madhu-1 commented on scripts/rook.sh:
```diff
@@ -2,7 +2,8 @@
 ROOK_DEPLOY_TIMEOUT=${ROOK_DEPLOY_TIMEOUT:-300}
-ROOK_URL="https://raw.githubusercontent.com/rook/rook/${ROOK_VERSION}/cluster/examples/kubernetes/ceph"
+ROOK_URL="https://raw.githubusercontent.com/rook/rook/${ROOK_VERSION}/"
+ROOK_DEPLOYMENT_PATH="cluster/examples/kubernetes/ceph"
```
@travisn operator logs are here https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/upgrade-tests-cephfs/detail/upgrade-tests-cephfs/4449/pipeline/99/. I tested this change locally everything is working fine not sure whats wrong.

Madhu-1 replied:
we don't use CSI provided by Rook to do any testing, we are making use of Rook to create a ceph cluster and will consume the ceph cluster from standalone cephcsi pods.

Madhu-1 replied:
nothing special here, the only change is this PR is changing the file paths as it's changed from 1.6.2 to 1.8.2, and the test is running fine locally. @travisn any other logs, we can check what is wrong here?

Madhu-1 commented:
@mergifyio rebase

Rakshith-R commented on scripts/rook.sh:
```diff
@@ -33,57 +34,67 @@ rook_version() {
+		curl -o "${TEMP_DIR}"/cluster-test.yaml "${ROOK_URL}/cluster-test.yaml"
+		sed -i "s|image.*|${ROOK_CEPH_CLUSTER_VERSION_IMAGE_PATH}|g" "${TEMP_DIR}"/cluster-test.yaml
+		sed -i "s/config: |/config: |\n    \[mon\]\n    mon_warn_on_insecure_global_id_reclaim_allowed = false/g" "${TEMP_DIR}"/cluster-test.yaml
+		sed -i "s/healthCheck:/healthCheck:\n    livenessProbe:\n      mon:\n        disabled: true\n      mgr:\n        disabled: true\n      mds:\n        disabled: true/g" "${TEMP_DIR}"/cluster-test.yaml
```
```suggestion
		sed -i "s/healthCheck:/healthCheck:\n    livenessProbe:\n      mon:\n        disabled: true\n      mgr:\n        disabled: true\n      mds:\n        disabled: true\n    startupProbe:\n      mon:\n        disabled: true\n      mgr:\n        disabled: true\n      mds:\n        disabled: true/g" "${TEMP_DIR}"/cluster-test.yaml
```

Madhu-1 replied:
added a commit, for now, let's see. if CI passes will update last commit to fix DCO and commitlint

Rakshith-R reviewed (commented):
startupprobe seems to be failing for mons, lets try disabling it. 
```
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  15m                   default-scheduler  Successfully assigned rook-ceph/rook-ceph-mon-a-6978b54fcc-jnqjf to minikube
  Normal   Pulled     15m                   kubelet            Container image "quay.io/ceph/ceph:v16" already present on machine
  Normal   Created    15m                   kubelet            Created container chown-container-data-dir
  Normal   Started    15m                   kubelet            Started container chown-container-data-dir
  Normal   Pulled     15m                   kubelet            Container image "quay.io/ceph/ceph:v16" already present on machine
  Normal   Created    15m                   kubelet            Created container init-mon-fs
  Normal   Started    15m                   kubelet            Started container init-mon-fs
  Normal   Pulled     14m (x2 over 15m)     kubelet            Container image "quay.io/ceph/ceph:v16" already present on machine
  Normal   Killing    14m                   kubelet            Container mon failed startup probe, will be restarted
  Normal   Created    14m (x2 over 15m)     kubelet            Created container mon
  Normal   Started    14m (x2 over 15m)     kubelet            Started container mon
  Warning  Unhealthy  10m (x26 over 15m)    kubelet            Startup probe failed:
  Warning  BackOff    52s (x31 over 8m54s)  kubelet            Back-off restarting failed container
  ```

Madhu-1 committed:
deploy: disable startupProbe

As startup probe is failing on the ceph pods,
disabling startupProbe in ceph cluster CRD.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
deploy: use osd as failureDomain for ec pool

in Rook 1.8 templates the failureDomain is
set to host for ec pools, as we are using
single node minikube cluster, setting the
failureDomain for osd.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
added some more checks for failureDomain for ec pools. hoping CI will pass :crossed_fingers:

Madhu-1 committed:
e2e: update cephfs data pool name

cephfs data pool name is changed from filesystem-data0
to filesystem-replicated in Rook 1.8. updating
the cephcsi helper functions also to use new
pool names.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
@mergifyio refresh

nixpanic commented:
@mergifyio refresh

nixpanic commented:
@mergifyio refresh

nixpanic commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

nixpanic commented:
@mergifyio refresh

nixpanic commented:
> /retest ci/centos/mini-e2e-helm/k8s-1.21

```
Jan 26 20:49:15.282: FAIL: backend images not matching kubernetes resource count,image count 22 kubernetes resource count 21
```
[logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.21/detail/mini-e2e-helm_k8s-1.21/2628/pipeline)

Madhu-1 commented:
@mergifyio refresh
